trendmicro_deepsecurity/icon_trendmicro_deepsecurity/actions/search_rules/action.py

```diff
@@ -0,0 +1,87 @@
+import komand
+from .schema import SearchRulesInput, SearchRulesOutput, Input, Output, Component
+# Custom imports below
+
+#from komand.exceptions import PluginException
+
+import requests
+import json
+
+
+from icon_trendmicro_deepsecurity.util.shared import tryJSON
+from icon_trendmicro_deepsecurity.util.shared import checkResponse
+
+class SearchRules(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='search_rules',
+                description=Component.DESCRIPTION,
+                input=SearchRulesInput(),
+                output=SearchRulesOutput())
+
+    def run(self, params={}):
+        '''
+        Searches for IPS rules by CVE number in Deep Security
+        '''
+
+        # Get parameters
+        self.vulnerabilities=params.get(Input.VULNERABILITIES)
+
+        matched_cves=set()
+        missed_cves=set()
+        ips_rules=set()    
+
+        for cve in self.vulnerabilities:
+
+            # Prepare request
+            url = f"{self.connection.dsm_url}/api/intrusionpreventionrules/search"
+
+            data = { "maxItems": 100,
+                     "searchCriteria": [ { "fieldName": "CVE",
+                                           "stringWildcards": True,
+                                           "stringValue": f"%{cve}%"
+                                         } ] }
+
+            post_header = {
+                            "Content-type": "application/json",
+                            "api-secret-key": self.connection.dsm_api_key,
+                            "api-version": "v1"
+                            }
+
+            # Search for IPS rules
+            response = requests.post(url,
+                                     data=json.dumps(data),
+                                     headers=post_header,
+                                     verify=True)
+            response.close()
+            
+            self.logger.info(cve)
+            self.logger.info(f'status: {response.status_code}')
```

Comment:
Keep quotation style consistent with the rest of the project (use double quotes)

Reply:
done